subsys/bluetooth/mesh/adv_ext.c

```diff
@@ -166,6 +167,9 @@ static void send_pending_adv(struct k_work *work)
 
 		BT_MESH_ADV(buf)->busy = 0U;
 		err = buf_send(buf);
+
+		net_buf_unref(buf);
```

Comment:
This looks wrong to me since `buf_send()` doesn't call `net_buf_ref()`

Reply:
Why `adv_start` cannot call `net_buf_ref`?

Reply:
Is this to your taste?

Reply:
Yes that makes more sense. And I hope it's more to your taste too: reference counting can be an ample source of bugs, so following some basic principles when using it makes the code more predictable and understandable.